crates/resolver/src/name_server/connection_provider.rs

```diff
@@ -94,6 +88,15 @@ pub trait RuntimeProvider: Clone + Send + Sync + Unpin + 'static {
         local_addr: SocketAddr,
         server_addr: SocketAddr,
     ) -> Pin<Box<dyn Send + Future<Output = io::Result<Self::Udp>>>>;
+
+    /// Create a UDP socket for QUIC usage.
+    fn bind_quic(
+        &self,
+        _local_addr: SocketAddr,
+        _server_addr: SocketAddr,
+    ) -> Result<Arc<dyn quinn::AsyncUdpSocket>, io::Error> {
+        Err(io::Error::from(io::ErrorKind::InvalidInput))
```

Comment:
This implementation is mainly there so that the async-std-resolver (which doesn't support any TLS paths today) can still implement this trait. Guarding this with features gets ugly. Not sure if we should have some associated const `SUPPORTS_QUIC: bool` so that we can preemptively make sure not to call this?